### DISCUSSION TITLE
Can I create groups of columns to be filtered by a tab system?

### DISCUSSION THREAD

**Discussion by jcc5018** [filamentphp/filament#11921] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

The only tabs functionality I see in the docs seem to be to create quick filters of data.

This is not what i need.

I have a few models in which i have a ton of columns. Lets just say 100 for sake of illustration. These columns should be grouped based on the date they represent. I have for instance a set of coumns related to ratings of the given model, and another related to demographics of the people who like a given model, or to display the number of other models connected to the parent model. 

So a different groupings of data, but could be queried as if one big table.  I need the data displayed in the table format for easy comparison, and filtering. 

Sample:
![image](https://github.com/filamentphp/filament/assets/21000134/a91c15de-36ca-4ccf-b142-a92783ea2650)

I would like to set up Column groups that can be toggled with the tab system. 
Individual columns can already be toggled, but how do I transfer that functionality to a group. And set it so if one group is visible, the others are hidden?  

public static function table(Table $table): Table
   {
       return $table
           ->columns([
     
               TextColumn::make('name')
                         ->searchable()
                         ->sortable(),
                TextColumn::make('column2')
                         ->searchable()
                         ->sortable(),
        TABGROUP::make('group_1")
                     ->columns([
                         ...6 more columns
                         ]),
        TABGROUP::make('group_2")
                     ->columns([
                         ...6 more columns
                         ])  ,
                         
                         Actions::   ...

..etc.

In this set up, the first 2 columns would display on every tab selection such as a frozen column in EXCEL
but each tab would display additional columns while maintaining the sort order or pagination of the given model

trailing columns such as actions could also be added as long as it is outside the TABGROUP

so in short, all table columns are the same but instead of one huge table where one might need to scroll, columns within TABGROUPS can be switched out for a different set. Any filters applied will search the entire table as if all data columns were displayed.
